src/MonitorProfiler/Environment/EnvironmentHelper.cpp

```diff
@@ -22,3 +22,33 @@ HRESULT EnvironmentHelper::SetProductVersion(
 
     return S_OK;
 }
+
+HRESULT EnvironmentHelper::GetRuntimeInstanceId(const std::shared_ptr<IEnvironment>& pEnvironment, const std::shared_ptr<ILogger>& pLogger, tstring& instanceId)
+{
+    HRESULT hr = S_OK;
+
+    IfFailLogRet(pEnvironment->GetEnvironmentVariable(s_wszRuntimeInstanceEnvVar, instanceId));
+
+    return S_OK;
+}
+
+HRESULT EnvironmentHelper::GetTempFolder(const std::shared_ptr<IEnvironment>& pEnvironment, const std::shared_ptr<ILogger>& pLogger, tstring& tempFolder)
+{
+    HRESULT hr = S_OK;
+
+    tstring tmpDir;
+#if TARGET_WINDOWS
+    IfFailLogRet(pEnvironment->GetEnvironmentVariable(s_wszTempEnvVar, tmpDir));
+#else
+    hr = pEnvironment->GetEnvironmentVariable(s_wszTempEnvVar, tmpDir);
+#endif
+
+    if (FAILED(hr))
+    {
+        tmpDir = s_wszDefaultTempFolder;
```

Comment:
s_wszDefaultTempFolder is `/tmp` on all platforms as the code is right now. This fallback should probably only be done for non-Windows platforms.

Reply:
I can address this on the next PR